### ISSUE TITLE
TextOverflow ellipsis with TextStyle height 1 cut off the letter

### ISSUE THREAD

**Issue by quyenvsp** [flutter/flutter#153988]
### Steps to reproduce

1. `flutter create .`
1. apply code sample
1. see result

### Expected results

Text should show clear, full height

### Actual results

Text cut off under baseline

### Code sample

<details open><summary>Code sample</summary>

```dart
const Text(
  'gggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg',
  overflow: TextOverflow.ellipsis,
  style: TextStyle(height: 1),
),
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

![Screenshot_1724393819](https://github.com/user-attachments/assets/f60be61b-bca6-4f66-8648-00c699c0d985)

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel master, 3.25.0-1.0.pre.102, on Microsoft Windows [Version 10.0.22631.4037], locale en-US)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 35.0.0)
[√] Chrome - develop for the web
[X] Visual Studio - develop Windows apps
    X Visual Studio not installed; this is necessary to develop Windows apps.
      Download at https://visualstudio.microsoft.com/downloads/.
      Please install the "Desktop development with C++" workload, including all of its default components
[√] Android Studio (version 2024.1)
[√] VS Code, 64-bit edition (version 1.92.2)
[√] Connected device (5 available)
[√] Network resources

! Doctor found issues in 1 category.
```

</details>

**Comment by huycozy** [comment]
Thanks for your report. This issue appears on stable 3.24.1 and previous stable release 3.22.3 as well.

**Comment by LongCatIsLooong** [comment]
> TextStyle(height: 1)

I think this is the expected behavior.  Setting height to `1` may result in clipping (same as setting `height` to something like 0.5) because you're setting the line-height to `UPEM` and some fonts don't guarantee that `UPEM == ascent + descent`. 
In this case the font has a negative internal line gap at the bottom. 

See: https://main-api.flutter.dev/flutter/painting/TextStyle/height.html

But it does sound weird that `ellipsis` applies clipping.

**Comment by quyenvsp** [comment]
Yes I know `height: 1` is expected behavior about line-height, but `ellipsis` should not clipping in this case.
And `height: 1` also is forced in _InputDecoratorState https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/input_decorator.dart#L2061

It make TextField label will cut off with default flutter font.

**Comment by jason-simmons** [comment]
Similar to https://github.com/flutter/flutter/issues/96322

**Comment by HE-LU** [comment]
I ran into this same issue today. Surprised it hasn't gotten more traction, given how critical a UI bug this is.
